src/ui/OwlUI.ts

```diff
@@ -47,28 +53,56 @@ export class OwlUI {
         };
     }
     /**
-     * Mount all of a plugin's Components
+     * Mount all of a plugin's Components to Owl and register its components'
+     * intents, actions and commands (TODO: this does not work for
+     * sub-components since those are mounted by their parent and we do not have
+     * access to their instantiation).
      *
      * @param {JWOwlUIPlugin} pluginInstance
      * @returns {Promise<void>}
      */
-    async _mountPluginComponents(pluginInstance: JWOwlUIPlugin): Promise<void> {
+    async _handleComponents(pluginInstance: JWOwlUIPlugin): Promise<void> {
         const env: PluginEnv = await this._createPluginEnv(pluginInstance);
-        const components = pluginInstance.componentsRegistry.slice();
-        components.forEach((_Component: typeof Component) => {
-            this._mount(_Component, env);
-        });
+        const Components = pluginInstance.componentsRegistry.slice();
+        for (let i = 0; i < Components.length; i++) {
+            const component = new Components[i](env);
+            await this._mount(component);
+            this._register(component, pluginInstance);
+        }
     }
     /**
-     * Mount a single component with the given `env`
+     * Mount a single component
      *
-     * @param {typeof OwlUIComponent} _Component
-     * @param {PluginEnv} env
+     * @param {OwlUIComponent} OwlUIComponent
      * @returns {Promise<void>}
      */
-    async _mount(_Component: typeof Component, env: PluginEnv): Promise<void> {
-        const component = new _Component(env);
+    async _mount(component: OwlUIComponent): Promise<void> {
         const target: HTMLElement = this.editor.el;
         await component.mount(target);
     }
+    /**
+     * Register a component's dispatcher registry records ('intents',
+     * 'commands', 'actions') in its plugin so it can in turn register them in
+     * Jabberwock's dispatcher
+     *
+     * @param {OwlUIComponent} component
+     * @param {JWOwlUIPlugin} pluginInstance
+     */
+    _register(component: OwlUIComponent, pluginInstance: JWOwlUIPlugin): void {
+        ['intents', 'actions', 'commands'].forEach((type: string): void => {
+            Object.keys(component[type]).forEach((name: string) => {
+                if (pluginInstance[type][name]) {
+                    console.warn(
+                        'Two components respond to the ' +
+                            type +
+                            ' "' +
+                            name +
+                            '". TODO: resolve conflict (chain commands if they exist).',
+                    );
+                } else {
+                    pluginInstance[type][name] = component[type][name];
+                }
+            });
+        });
+    }
```

Comment:
For the moment, I would move this inside the DevTools plugin. I think it is the very special case of the DevTools which wants to be a UI plugin AND respond to intents/actions/etc in the same time. I would not offer this possibility to other UI plugins in the core system (yet.)

Reply:
I guess you're right.

Reply:
Damn I think I can't. The component has to be mounted and that is done in OwlUI.

Reply:
OK I got it to work. Only trade-off: now a JWOwlUIPlugin has to manually call an init() method in its constructor. That is in order to mount its own specific components. I can't automize that in the parent class because that would imply initializing the Components list before calling super(), which is not supported by typescript. But I feel that's a very minor trade-off, and that's allowed me to bypass it in DevTools so I could register the handlers _before_ mounting the components.